Library/Homebrew/test/requirement_spec.rb

```diff
@@ -0,0 +1,237 @@
+require "requirement"
+
+RSpec::Matchers.alias_matcher :have_a_default_formula, :be_a_default_formula
+
+describe Requirement do
+  subject { klass.new }
+
+  describe "#tags" do
+    subject { described_class.new(tags) }
+
+    context "single tag" do
+      let(:tags) { ["bar"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "multiple tags" do
+      let(:tags) { ["bar", "baz"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "symbol tags" do
+      let(:tags) { [:build] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+
+    context "symbol and string tags" do
+      let(:tags) { [:build, "bar"] }
+
+      its(:tags) { are_expected.to eq(tags) }
+    end
+  end
+
+  describe "#fatal?" do
+    subject { Class.new(described_class) { fatal true }.new }
+
+    it { is_expected.to be_fatal }
+  end
+
+  describe "#satisfied?" do
+    context "#satisfy with block and build_env returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy(build_env: false) do
+            true
+          end
+        end
+      end
+
+      it { is_expected.to be_satisfied }
+    end
+
+    context "#satisfy with block and build_env returns false" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy(build_env: false) do
+            false
+          end
+        end
+      end
+
+      it { is_expected.not_to be_satisfied }
+    end
+
+    context "#satisfy returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy true
+        end
+      end
+
+      it { is_expected.to be_satisfied }
+    end
+
+    context "#satisfy returns false" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy false
+        end
+      end
+
+      it { is_expected.not_to be_satisfied }
+    end
+
+    context "#satisfy with block returns true" do
+      let(:klass) do
+        Class.new(described_class) do
+          satisfy { true }
+        end
+      end
+
+      it "sets up build environment by default" do
+        expect(ENV).to receive(:with_build_environment) do
+          yield true if block_given?
+          true
+        end
+
+        expect(subject).to be_satisfied
```

Comment:
This is really two different tests, isn't it? One testing the return value as above, and another testing the ENV message.